### PR TITLE
executor: support batch commit the DMLs (#9882)

### PR DESCRIPTION
**Pull request by jackysp**
Conflicts:
	executor/delete.go
	executor/errors.go
	executor/insert_common.go
	executor/simple.go
	executor/update.go
	executor/write_test.go
	session/session.go

### What problem does this PR solve? 
Cherry-pick #9882 to release 3.0. Be careful, many conflicts...

### CODE REVIEW

File: executor/insert_common.go [hunk 1]
```diff
@@ -365,7 +364,7 @@ func (e *InsertValues) doBatchInsert(ctx context.Context) error {
 	}
 	if err := e.ctx.NewTxn(ctx); err != nil {
 		// We should return a special error for batch insert.
-		return ErrBatchInsertFail.GenWithStack("BatchInsert failed with error: %v", err)
+		return ErrBatchDMLFail.GenWithStack("BatchInsert failed with error: %v", err)
```
Comment (lysu) [inline executor/insert_common.go]: where does `doBatchInsert` will be called?

**Comment by coocood** [comment]
LGTM

File: executor/insert_common.go (hunk 1 above)
Comment (jackysp) [reply executor/insert_common.go]: It should be useless. I'll remove it.

Comment (jackysp) [reply executor/insert_common.go]: addressed, PTAL

File: executor/write.go [hunk 2]
```diff
@@ -188,3 +188,15 @@ func getTableOffset(schema *expression.Schema, handleCol *expression.Column) int
...
+
+func batchDMLCommit(gctx context.Context, ctx sessionctx.Context) error {
+	if err := ctx.StmtCommit(); err != nil {
+		return ErrBatchDMLFail.GenWithStackByArgs(err)
+	}
+	ctx.GetSessionVars().TxnCtx.IsBatched = true
+	if err := ctx.NewTxn(gctx); err != nil {
+		return ErrBatchDMLFail.GenWithStackByArgs(err)
+	}
+	ctx.GetSessionVars().TxnCtx.IsBatched = true
```
Comment (tiancaiamao) [inline executor/write.go]: Repeated? see line 196

Comment (jackysp) [reply executor/write.go]: The `TxnCtx` is refreshed at line 196, so `IsBatched` is set once again.

Comment (tiancaiamao) [reply executor/write.go]: Get it

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**